lib/patch-series.ts

```diff
@@ -468,7 +468,7 @@ export class PatchSeries {
             }
 
             header = authorMatch[1] + replaceSender + authorMatch[3];
-            if (i === 0 && senderName) {
+            if (mails.length > 1 && i === 0 && senderName) {
```

Comment:
Does this also need to be taken into account when setting `onBehalfOf` at line 456?

Reply:
No (I tried to explain that in the commit message, but admittedly that explanation is a bit convoluted).

Imagine that you write a patch, open a PR at git-for-windows/git and I merge it. The implicit agreement is that it is now _my_ responsibility to shepherd that patch into Git. So I open a PR at gitgitgadget/git. You are the author, I am the sender. The patch should still be sent on my behalf, but the `From:` line with your authorship should be prepended to the mail. Read: `onBehalfOf` is still correct (at least in my opinion).